### PR TITLE
Update angular-file-upload definitions

### PR DESCRIPTION
**Pull request by dalie**
Added missing methods to the upload promise.

### CODE REVIEW

File: angular-file-upload/angular-file-upload-tests.ts [hunk 1]
```diff
@@ -29,9 +29,13 @@ module controllers {
                     },
                     file: file
                 })
+                .abort()
```
Comment (StephaneAlie) [inline angular-file-upload/angular-file-upload-tests.ts]: Should we do the .then with success / error stuff using the IUploadPromise return value of abort?

Comment (dalie) [inline angular-file-upload/angular-file-upload-tests.ts]: There is only one "then", it's the same promise.

Comment (StephaneAlie) [inline angular-file-upload/angular-file-upload-tests.ts]: Ah yeah, self returning chainable thing a magig

**Comment by StephaneAlie** [comment]
LGTM

**Comment by AnthonyMacKinnon** [comment]
LGTM

**Pull request merged**